### PR TITLE
Done

### PR DESCRIPTION
**Pull request by aionova**
[DEMO LINK](https://aionova.github.io/react_clock/)

### CODE REVIEW

**Review by yazheviks — APPROVED** [review]
Класс!)

File: src/App.js [hunk 1]
```diff
@@ -1,25 +1,57 @@
...
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
+class App extends React.Component {
+  state = {
+    isClockVisible: true,
+    name: 0,
+  };
+
+  toggleVisible = () => {
```
Comment (aliceblunt8) [inline src/App.js:11]:
```suggestion
  toggleVisibility = () => {
```

File: src/components/Clock/Clock.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  componentDidMount() {
+    setInterval(() => {
+      this.setState({
+        time: (new Date()).toLocaleTimeString(),
+      });
+      // eslint-disable-next-line no-console
+      console.log(new Date().toLocaleTimeString());
+    }, 1000);
+  }
```
Comment (aliceblunt8) [inline src/components/Clock/Clock.js:17]: you also need to clear the timer when the Clock is removed from the DOM, cause now the clock is still ticking in the console event after the Clock component is destroyed
hint: use componentWillMount and clearInterval
https://reactjs.org/docs/state-and-lifecycle.html

File: src/components/Clock/Clock.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return (
+      <div className="clock">
+        <h1>
+          {`Clock named ${this.props.name}`}
+        </h1>
+        <p>
+          {this.props.isClockVisible ? `Current time:
+          ${this.state.time}` : 'No clock available'}
+        </p>
+      </div>
```
Comment (aliceblunt8) [inline src/components/Clock/Clock.js:29]: it's better to destructure state and props

File: src/components/Clock/Clock.js (hunk 2 above)
Comment (aliceblunt8) [reply src/components/Clock/Clock.js:17]: also, let's create a separate method for updating the time (we can call it 'tick')

**Review by aliceblunt8 — CHANGES_REQUESTED** [review]

**Review by aliceblunt8 — APPROVED** [review]

**Pull request closed without merging**